sources/web/datalab/polymer/components/resizable-divider/resizable-divider.ts

```diff
@@ -79,40 +99,77 @@ class ResizableDividerElement extends Polymer.Element {
 
     // stop styling the handle as active because dragging is done
     this.$.divider.classList.remove('active');
+    this._lastMouseUpPosition = this.dividerPosition;
   }
 
   _mouseMoveHandler(event: MouseEvent) {
-    this._resizePanes(event.clientX);
+    const container = this.$.container as HTMLDivElement;
+    const containerRect = container.getBoundingClientRect();
+    const newLeftPaneWidth = event.clientX - containerRect.left;
+    const fullWidthLessDivider = containerRect.width - this._dividerWidth;
+    const newPosition = newLeftPaneWidth * 100 / fullWidthLessDivider;
+    this.dividerPosition =
+        (newPosition < 0) ? 0 : (newPosition > 100) ? 100 : newPosition;
+    // Let observer call _dividerMoved
+  }
+
+  /**
+   * Calculate the new divider position after hideRight changes.
+   */
+  _hideChanged() {
+    if (!this._readyDone) {
+      return;   // Leave divider position unchanged on startup
+    } else if (this.hideRight) {
+      this.dividerPosition = 100;
+    } else {
+      // Make the right pane visible
+      if (this._lastMouseUpPosition > 0 &&
+          this._lastMouseUpPosition < 100) {
+        this.dividerPosition = this._lastMouseUpPosition;
+      } else if (this._lastMouseDownPosition > 0 &&
```

Comment:
Curious when does this happen?

Reply:
The mouse-up value is used as long as the user has not dragged the divider all the way over to close the pane; but if that did happen, then the mouse-down value is used, i.e. where the user dragged from the last time.

Reply:
Ah makes sense.